sycl/include/CL/sycl/handler.hpp

```diff
@@ -1060,21 +1091,32 @@ class handler {
                   "Invalid source accessor target for the copy method.");
     static_assert(isValidTargetForExplicitOp(AccessTarget_Dst),
                   "Invalid destination accessor target for the copy method.");
-#ifndef __SYCL_DEVICE_ONLY__
-    if (MIsHost) {
-      range<Dims_Src> Range = Dst.get_range();
+    // TODO replace to get_size() when it will provide correct values.
+    assert(
+        (Dst.get_range().size() * sizeof(T_Dst) >=
+         Src.get_range().size() * sizeof(T_Src)) &&
+        "dest must have at least as many bytes as the range accessed by src.");
+    if (MIsHost ||
+        !isCopyingBySliceAvailable(Src.get_range(), Dst.get_range())) {
```

Comment:
Not sure I understand what isCopyingBySliceAvailable means and how a copy algorithm (will) differs depending on this check. Could you please clarify that?

Reply:
We can copy by slice(`clEnqueueCopyBufferRect`) under some conditions, for example:
case 2x2 to 3x3

```
12
34
```
->
```
000
000
000
```
 =
```
120
340
000
```

and in this case we can't (the source slice exceeds the destination slice)
case 4x2 to 2x4

```
1234
5678
```
->
```
00
00
00
00
```
=
```
1234
5678
00
00
```
but we can copy sequentially.
```
12
34
56
78
```
I plan to create an issue to describe all valid cases allowed for the copy operation and update CTS. The CTS do not cover all cases completely.